localstack/services/ec2/resource_providers/aws_ec2_transitgateway.py

```diff
@@ -0,0 +1,145 @@
+    Value: Optional[str]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class EC2TransitGatewayProvider(ResourceProvider[EC2TransitGatewayProperties]):
+
+    TYPE = "AWS::EC2::TransitGateway"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[EC2TransitGatewayProperties],
+    ) -> ProgressEvent[EC2TransitGatewayProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/Id
+
+
+
+        Create-only properties:
+          - /properties/AmazonSideAsn
+          - /properties/MulticastSupport
+
+        Read-only properties:
+          - /properties/Id
+
+        IAM permissions required:
+          - ec2:CreateTransitGateway
+          - ec2:CreateTags
+
+        """
+        model = request.desired_state
+        create_params = {
+            "Options": util.select_attributes(
+                model,
+                [
+                    "AmazonSideAsn",
+                    "AssociationDefaultRouteTableId",
+                    "AutoAcceptSharedAttachments",
+                    "DefaultRouteTableAssociation",
+                    "DefaultRouteTablePropagation",
+                    "DnsSupport",
+                    "MulticastSupport",
+                    "PropagationDefaultRouteTableId",
+                    "TransitGatewayCidrBlocks",
+                    "VpnEcmpSupport",
+                ],
+            )
+        }
+
+        if model.get("Description"):
+            create_params["Description"] = model["Description"]
+
+        if model.get("Tags", []):
+            create_params["TagSpecifications"] = [
+                {"ResourceType": "transit-gateway", "Tags": model["Tags"]}
+            ]
+
+        response = request.aws_client_factory.ec2.create_transit_gateway(**create_params)
+        model["Id"] = response["TransitGateway"]["TransitGatewayId"]
+
+        return ProgressEvent(status=OperationStatus.SUCCESS, resource_model=model)
+
+    def read(
+        self,
+        request: ResourceRequest[EC2TransitGatewayProperties],
+    ) -> ProgressEvent[EC2TransitGatewayProperties]:
+        """
+        Fetch resource information
+
+        IAM permissions required:
+          - ec2:DescribeTransitGateways
+          - ec2:DescribeTags
+        """
+        raise NotImplementedError
+
+    def delete(
+        self,
+        request: ResourceRequest[EC2TransitGatewayProperties],
+    ) -> ProgressEvent[EC2TransitGatewayProperties]:
+        """
+        Delete a resource
+
+        IAM permissions required:
+          - ec2:DeleteTransitGateway
+          - ec2:DeleteTags
+        """
+        model = request.desired_state
+        request.aws_client_factory.ec2.delete_transit_gateway(TransitGatewayId=model["Id"])
+        return ProgressEvent(
+            status=OperationStatus.SUCCESS,
+            resource_model={},
```

Comment:
I don't understand how this does not fail any tests. We look at the properties of the `resource_model` on execution of the function and extract fields from it. Can you double check that there are no messages in the logs about "failed to delete..." or similar?

Reply:
I extended the test to assert the deletion. It seems that for the TransitGateway, AWS and moto act different but no error was thrown during the deletion.